File: backend/app/schemas/checklist.py
from typing import Literal

from pydantic import BaseModel

ChecklistTag = Literal["positive", "negative", "earnings", "caution", "neutral"]


class ChecklistItem(BaseModel):
    id: str
    tag: ChecklistTag
    headline: str
    description: str
    source_count: int
    url: str = ""


class ChecklistResponse(BaseModel):
    ticker: str
    date: str
    total_article_count: int
    items: list[ChecklistItem]

File: frontend/src/shared/types/checklist.ts
export type ChecklistTag = "positive" | "negative" | "earnings" | "caution" | "neutral";

export interface ChecklistItem {
  id: string;
  tag: ChecklistTag;
  headline: string;
  description: string;
  source_count: number;
  url: string;
}

export interface ChecklistResponse {
  ticker: string;
  date: string;
  total_article_count: number;
  items: ChecklistItem[];
}
